migration-engine/connectors/sql-migration-connector/src/flavour/postgres.rs

```diff
@@ -264,14 +264,25 @@ impl SqlFlavour for PostgresFlavour {
         })
     }
 
-    fn reset(&mut self) -> BoxFuture<'_, ConnectorResult<()>> {
+    fn reset(&mut self, namespaces: Option<Namespaces>) -> BoxFuture<'_, ConnectorResult<()>> {
         with_connection(self, move |params, _circumstances, conn| async move {
-            let schema_name = params.url.schema();
+            let mut schemas = vec![Cow::Borrowed(params.url.schema())];
 
-            conn.raw_cmd(&format!("DROP SCHEMA \"{}\" CASCADE", schema_name), &params.url)
-                .await?;
-            conn.raw_cmd(&format!("CREATE SCHEMA \"{}\"", schema_name), &params.url)
-                .await?;
+            // We reset the namespaces defined in the `schemas` datasource property _in addition to
+            // the schema in the search path_, because the search path is where we create the
+            // migrations table.
+            for namespace in namespaces.into_iter().flatten() {
+                schemas.push(Cow::Owned(namespace))
+            }
+
+            tracing::info!(?schemas, "Resetting schema(s)");
+
+            for schema_name in schemas {
+                conn.raw_cmd(&format!("DROP SCHEMA \"{}\" CASCADE", schema_name), &params.url)
+                    .await?;
+                conn.raw_cmd(&format!("CREATE SCHEMA \"{}\"", schema_name), &params.url)
+                    .await?;
+            }
```

Comment:
Does this mean that if the user has a schema defined in their connection string, we reset that on top of the schemas defined in the schema file?

Do we warn before at least? That sounds like it could potentially be disastrous if the user isn't careful.

Reply:
That's why we wanted to take time to think this through. I don't have an answer to that. What this does with multiSchema is the same as without multiSchema, just adding the namespaces defined in the Prisma schema